--- /home/ubuntu/libffi-rs/libffi-sys-rs/libffi/src/x86/ffi.c	2026-08-12 06:32:31.020816588 +0000
+++ /home/ubuntu/work/translation-input/x86/ffi.c	2026-08-12 06:41:50.227187551 +0000
@@ -251,15 +251,9 @@
   [FFI_MS_CDECL] = { 1, R_ECX, 0 }
 };
 
-#ifdef HAVE_FASTCALL
-  #ifdef _MSC_VER
-    #define FFI_DECLARE_FASTCALL __fastcall
-  #else
-    #define FFI_DECLARE_FASTCALL __declspec(fastcall)
-  #endif
-#else
-  #define FFI_DECLARE_FASTCALL
-#endif
+/* C2Rust 0.22.1 cannot export Clang AttributedType fastcall nodes.
+   Translation only: restore extern "fastcall" in generated Rust. */
+#define FFI_DECLARE_FASTCALL
 
 extern void FFI_DECLARE_FASTCALL ffi_call_i386(struct call_frame *, char *) FFI_HIDDEN;
 
